### PR TITLE
make end work under Julia 0.7

### PR DESCRIPTION
**Pull request by bkamins**
following issue #1370

### CODE REVIEW

File: test/dataframe.jl [hunk 1]
```diff
@@ -582,4 +582,32 @@ module TestDataFrame
         df[4] = [1,2,3]
         @test names(df) == [:x3, :x3_1, :x3_2, :x4]
     end
+
+    @testset "handling of end in indexing" begin
+        z = DataFrames.DataFrame(rand(4,5))
```
Comment (nalimilan) [inline test/dataframe.jl]: Why the `DataFrames` prefix? Same below.

Otherwise, looks good.

Comment (bkamins) [reply test/dataframe.jl]: leftover from initial testing. fixed.

**Comment by ExpandingMan** [comment]
Would be great if we could merge this as currently missing these methods is breaking all sorts of things in 0.7 :wink:.

**Comment by bkamins** [comment]
@ExpandingMan I would propose that you make a separate PR for `STDOUT` as it is an unrelated change to `end` so it will be better to keep it as a different commit to master. And you do not have to wait till this PR is merged.

**Pull request merged**